sections/collage.liquid

```diff
@@ -90,19 +90,63 @@
                 </div>
               </div>
             {%- when 'product' -%}
+              {% liquid
+                unless section.blocks.size == 1
+                  if forloop.first and section.settings.desktop_layout == 'left'
+                    assign columns_amount_desktop = 1.5
+                  elsif forloop.last and section.settings.desktop_layout == 'right'
+                    assign columns_amount_desktop = 1.5
+                  else
+                    assign columns_amount_desktop = 3
+                  endif
+                endunless
+
+                assign columns_amount_mobile = 1
+                unless section.settings.mobile_layout == 'column' or section.blocks.size == 1
+                  unless section.blocks.size == 3 and forloop.first and section.settings.desktop_layout == 'left'
+                    unless section.blocks.size == 3 and forloop.last and section.settings.desktop_layout == 'right'
+                      assign columns_amount_mobile = 2
+                    endunless
+                  endunless
+                endunless
+              %}
               {% render 'card-product',
                 card_product: block.settings.product,
                 media_aspect_ratio: 'adapt',
                 show_secondary_image: block.settings.second_image,
-                extend_height: true
+                extend_height: true,
+                columns_desktop: columns_amount_desktop,
+                columns_mobile: columns_amount_mobile
               %}
             {%- when 'collection' -%}
+              {% liquid
+                unless section.blocks.size == 1
+                  if forloop.first and section.settings.desktop_layout == 'left'
+                    assign columns_amount_desktop = 1.5
+                  elsif forloop.last and section.settings.desktop_layout == 'right'
+                    assign columns_amount_desktop = 1.5
+                  else
+                    assign columns_amount_desktop = 3
+                  endif
+                endunless
+
+                assign columns_amount_mobile = 1
+                unless section.settings.mobile_layout == 'column' or section.blocks.size == 1
+                  unless section.blocks.size == 3 and forloop.first and section.settings.desktop_layout == 'left'
+                    unless section.blocks.size == 3 and forloop.last and section.settings.desktop_layout == 'right'
+                      assign columns_amount_mobile = 2
+                    endunless
+                  endunless
+                endunless
+              %}
```

Comment:
I wonder if we need to repeat this twice 🤔 Since it just needs to be within the for loop. We could wrap it in an if statement `if block.type == 'collection' or block.type == 'product'` though that adds to the list of conditional already present.

Reply:
Yeah for this one I wasn't sure what to do; having the block of code only once does seem more appealing.

> it just needs to be within the for loop

My issue was "do we want to add an _if_ check when there's already a _case_ coming?
If we're fine with that I'm happy to move the logic before the case